Frontend/ui-driver/src/Flow.purs

```diff
@@ -2360,25 +2360,25 @@ homeScreenFlow = do
               })
           onSuccessEndRide
         Left errorPayload -> do
-          (GetRidesHistoryResp rideHistoryResponse) <- Remote.getRideHistoryReqBT "1" "0" "true" "null" "null"
-          case Tuple (head rideHistoryResponse.list) (isJust state.data.advancedRideData)  of
-            Tuple Nothing _-> do
-                onSuccessEndRide
-            Tuple (Just _) true -> do
-              void $ lift $ lift $ toggleLoader false
-              let errResp = errorPayload.response
-              let codeMessage = decodeErrorCode errResp.errorMessage
-              liftFlowBT $ logEvent logField_ "incorrect flow"
-              if ( errorPayload.code == 400 && codeMessage == "INCORRECT_OTP") then do
-                modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen { props {otpIncorrect = true, enterOtpModal = endRideOtpModalOnError, otpAttemptsExceeded = false, rideOtp = "",enterOdometerReadingModal= false, endRideOdometerReadingModal = false, enterOtpFocusIndex = 0, enterOdometerFocusIndex=0} })
-              else if ( errorPayload.code == 429 && codeMessage == "HITS_LIMIT_EXCEED") then do
-                modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen { props {otpIncorrect = false,otpAttemptsExceeded = true, enterOtpModal = endRideOtpModalOnError, rideOtp = "",enterOdometerReadingModal= false,endRideOdometerReadingModal = false, enterOtpFocusIndex = 0, enterOdometerFocusIndex=0} })
-              else do
-                pure $ toast $ getString SOMETHING_WENT_WRONG_PLEASE_TRY_AGAIN
-                homeScreenFlow
-            Tuple (Just _) false -> do
-              onSuccessEndRide
+          (GetRidesHistoryResp rideHistoryResponse) <- Remote.getRideHistoryReqBT "2" "0" "true" "null" "null"
+          case Tuple (DA.length $ rideHistoryResponse.list) (isJust state.data.advancedRideData)  of
+            Tuple 1 true -> onSuccessEndRide
+            Tuple 2 true -> errorHandlerEndRide errorPayload endRideOtpModalOnError
```

Comment:
pls don't merge reviewing edge case for this condition 
`Tuple 1 true -> onSuccessEndRide`

Reply:
edge case is if advanced ride is canceled at same instance and the ride api gives one current ride